### PR TITLE
ENG-215 [debug endpoint] Create dummy debug endpoint

### PR DESCRIPTION
**Pull request by Jiyuan-Bolt**
# Description
This PR creates a new web API which will be used by Bolt server to retrieve merchant server information for debugging purpose. For more details, please check Tech Spec: https://docs.google.com/document/d/1Q-0NZHQew1ALm4MOyVGl2m0-JUG_mnmhcn9zPcqJo98/edit#heading=h.cunzcjjb9kz

The response format will look like 
```
{
    "php_version": "todo",
    "platform_version": "todo",
    "bolt_config_settings": [
        {
            "name": null,
            "value": null
        }
    ],
    "other_plugin_versions": [
        {
            "name": null,
            "version": null
        }
    ]
}
```
Note: Logs are not included in the response yet, because there needs some more investigation on the best format. 

Fixes: [ENG-215]

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

#changelog create dummy debug endpoint

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

[ENG-215]: https://boltpay.atlassian.net/browse/ENG-215

### CODE REVIEW

File: etc/di.xml [hunk 1]
```diff
@@ -53,6 +53,12 @@
     <preference for="Bolt\Boltpay\Api\Data\FeatureSwitchInterface" type="Bolt\Boltpay\Model\FeatureSwitch"/>
     <preference for="Bolt\Boltpay\Api\FeatureSwitchRepositoryInterface" type="Bolt\Boltpay\Model\FeatureSwitchRepository"/>
 
+    <!-- For debug endpoint -->
+    <preference for="Bolt\Boltpay\Api\DebugInterface" type="Bolt\Boltpay\Model\Api\Debug" />
+    <preference for="Bolt\Boltpay\Api\Data\DebugInfoInterface" type="Bolt\Boltpay\Model\Api\Data\DebugInfo" />
+    <preference for="Bolt\Boltpay\Api\Data\PluginVersionInterface" type="Bolt\Boltpay\Model\Api\Data\PluginVersion" />
+    <preference for="Bolt\Boltpay\Api\Data\BoltConfigSettingInterface" type="Bolt\Boltpay\Model\Api\Data\BoltConfigSetting" />
```
Comment (roopakv) [inline etc/di.xml]: Why make these di classes as opposed to a simple Php class?

Comment (Jiyuan-Bolt) [reply etc/di.xml]: I actually just followed the same way as other places (eg: shipping methods endpoint) and thought it was the recommended way as Magento2 seems to really like DI.

@danac-gs @YevhenShBolt @matt-thomason @njfaries  What are your thoughts on these?

Comment (Jiyuan-Bolt) [reply etc/di.xml]: Synced on Slack and decided to only leave DebugInterface.

**Comment by Jiyuan-Bolt** [comment]
This PR only has a bunch of models with getters and setters only. The tests will be added when adding actual business logic.

File: etc/di.xml (hunk 1 above)
Comment (Jiyuan-Bolt) [reply etc/di.xml]: removed

**Comment by Jiyuan-Bolt** [comment]
> The DI classes seem like a lot of over head. Why not simple Php classes

Fixed. We only keep DebugInterface now.

File: Model/Api/Data/BoltConfigSetting.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	 * Set Bolt setting name.
+	 *
+	 * @api
+	 * @param string $name
+	 *
+	 * @return $this
+	 */
+	public function setName($name)
+	{
+		$this->name = $name;
```
Comment (matt-thomason) [inline Model/Api/Data/BoltConfigSetting.php]: Spacing seems really wide here - is your IDE set to tabs being 8 spaces?

Comment (Jiyuan-Bolt) [reply Model/Api/Data/BoltConfigSetting.php]: It's 4 space, not sure why github displays this way

**Pull request merged**